### PR TITLE
Move mysql interface to resources

### PR DESCRIPTION
**Pull request by rylev**
The same thing as #1886 but for mysql - this changes the mysql interface to use a connection oriented resource.

### CODE REVIEW

File: crates/outbound-mysql/src/lib.rs [hunk 1]
```diff
@@ -23,37 +32,48 @@ impl HostComponent for OutboundMysql {
...
+impl v2::HostConnection for OutboundMysql {
+    async fn open(&mut self, address: String) -> Result<Result<Resource<Connection>, v2::Error>> {
+        Ok(async {
+            self.connections
+                .push(
+                    build_conn(&address)
+                        .await
+                        .map_err(|e| v2::Error::ConnectionFailed(format!("{e:?}")))?,
+                )
+                .map_err(|_| v2::Error::Other("too many connections".into()))
```
Comment (lann) [inline crates/outbound-mysql/src/lib.rs:56]: Should this be a `ConnectionFailed` too?

**Review by lann — APPROVED** [review]

File: wit/preview2/mysql.wit [hunk 2]
```diff
@@ -0,0 +1,24 @@
+interface mysql {
+  use fermyon:spin/rdbms-types.{parameter-value, row-set}
+
+  /// Errors related to interacting with Mysql.
```
Comment (fibonacci1729) [inline wit/preview2/mysql.wit:4]: question / super nit: is it Mysql or MySQL?

**Review by fibonacci1729 — APPROVED** [review]

File: wit/preview2/mysql.wit [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+      connection-failed(string),
+      bad-parameter(string),
+      query-failed(string),
+      value-conversion-failed(string),
+      other(string)
+  }
+
+  /// A connection to a mysql database.
+  resource connection {
+    /// Open a connection to the Postgres instance at `address`.
```
Comment (michelleN) [inline wit/preview2/mysql.wit:15]: `Postgres` -> `MySQL`?

**Review by michelleN — CHANGES_REQUESTED** [review]

**Comment by rylev** [comment]
Merging as all the feedback has been addressed.

**Pull request merged**